Examples/SnapinsSDKKotlinExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/SupportHomeViewAddition.java

```diff
@@ -0,0 +1,236 @@
+                        break;
+                    case R.id.action_chat:
+                        launchChat();
+                        break;
+                    case R.id.action_sos:
+                        launchSos();
+                        break;
+                }
+                return true;
+            }
+        });
+
+        if (!visible) {
+            fab.setScaleX(0);
+            fab.setScaleY(0);
+        }
+
+        context = fab.getContext();
+
+    }
+
+
+    /**
+     * Determines which scenes should have the view addition.
+     */
+    @Override
+    public boolean visibleFor(KnowledgeScene knowledgeScene) {
+        return (knowledgeScene == KnowledgeScene.SCENE_HOME);
+    }
+
+    /**
+     * Returns the entrance animator object.
+     */
+    @NonNull
+    @Override
+    public Animator getEnterAnimator(View view) {
+        return getAnimatorSet(view, 1f);
+    }
+
+    /**
+     * Returns the exit animator object.
+     */
+    @NonNull
+    @Override
+    public Animator getExitAnimator(View view) {
+        return getAnimatorSet(view, 0f);
+    }
+
+    private Animator getAnimatorSet(View view, Float value){
+        View fab = view.findViewById(R.id.support_home_fab);
+
+        AnimatorSet set = new AnimatorSet();
+        set.playTogether(
+                ObjectAnimator.ofFloat(fab, "scaleX", value),
+                ObjectAnimator.ofFloat(fab, "scaleY", value)
+        );
+        return set;
+    }
+
+    /**
+     * Configures and launches Live Agent Chat
+     */
+    private void launchChat() {
+        // Create a UI configuration instance from a core config object
+        ChatConfiguration chatConfiguration = ServiceSDKUtils.getChatConfigurationBuilder(context)
+                .preChatFields(buildPreChatFields())
+                .build();
+
+        // Configure chat session listener
+        ServiceSDKApplication serviceSDKApplication = (ServiceSDKApplication) context.getApplicationContext();
+        final ChatSessionListener chatListener = serviceSDKApplication.getChatSessionListener();
+
+        ChatUI.configure(ChatUIConfiguration.create(chatConfiguration))
+                .createClient(context)
+                .onResult(new Async.ResultHandler<ChatUIClient>() {
+                    @Override
+                    public void handleResult(Async<?> async, @NonNull ChatUIClient chatUIClient) {
+                            // Add the configured chat session listener to the Chat UI client
+                            chatUIClient.addSessionStateListener(chatListener);
+                            // Start the live agent chat session
+                            chatUIClient.startChatSession((FragmentActivity) context);
+                    }
+                });
+    }
+
+    /**
+     * Configures pre chat fields if prechat is enabled in settings
+     */
+    private List<PreChatField> buildPreChatFields(){
+        PreChatField preChatField1 = new PreChatField.Builder()
```

Comment:
This line could use a comment, like:
`// Create a pre-chat field for the user's name`
Actually, I'm not sure that is accurate, but it would be nice to comment on exactly what you're doing with each pre-chat field... If it's a hidden field, specify that too...

Reply:
done